pontoon/base/models/user.py

```diff
@@ -433,6 +435,42 @@ def serialized_notifications(self):
     }
 
 
+def is_subscribed_to_notification(self, notification):
+    """
+    Determines if the user has email subscription to the given notification.
+    """
+    profile = self.profile
+    verb = notification.verb
+    desc = notification.description
+
+    # New strings notifications
+    if re.match(r"updated with \d+ new strings", verb):
+        return profile.new_string_notifications_email
```

Comment:
This reliance on the exact notification language _will_ break. Should instead add a new `category` or similar field to notifications that matches the applicable permission.

Reply:
We already rely on the value of the `verb` to categorize notifications when we render them.

The code would indeed be much less brittle if there was a dedicated category field on the Notification model, but we use an external library for handling notifications, and the model doesn't support that. 

Should we modify the code to send notifications to store {type: SOME-TYPE} in the [`Notification.data`](https://github.com/django-notifications/django-notifications/blob/master/notifications/base/models.py#L232) field and then query it here?

Reply:
> The code would indeed be much less brittle if there was a dedicated category field on the Notification model, but we use an external library for handling notifications, and the model doesn't support that.

Are you sure? Wasn't that support added in django-notifications/django-notifications#309?

Reply:
Yeah, yeah - I think we're talking about the same thing:
https://github.com/django-notifications/django-notifications?tab=readme-ov-file#extra-data

Reply:
No, take a look at that PR. It's about adding a proper custom data column, rather than using the `data` JSON text blob (which is not even stored as `jsonb`, pending on django-notifications/django-notifications#385).

Reply:
Oh, didn't realize this was possible:
https://github.com/django-notifications/django-notifications?tab=readme-ov-file#abstractnotification-model

Looking into it.

Reply:
Hm, I'm running out of ideas of why I'm hitting this error when trying to run a data migration for the new, custom model...

It might be simpler at this stage to just use the default model, and store the category in the data field. The only (small) downside would be that we'd need to use `__contais` to query it.

```
pontoon@d0168d836c37:/app$ ./manage.py showmigrations
SystemCheckError: System check identified some issues:

ERRORS:
base.Notification.action_object_content_type: (fields.E304) Reverse accessor 'ContentType.notify_action_object' for 'base.Notification.action_object_content_type' clashes with reverse accessor for 'notifications.Notification.action_object_content_type'.
	HINT: Add or change a related_name argument to the definition for 'base.Notification.action_object_content_type' or 'notifications.Notification.action_object_content_type'.
base.Notification.action_object_content_type: (fields.E305) Reverse query name for 'base.Notification.action_object_content_type' clashes with reverse query name for 'notifications.Notification.action_object_content_type'.
	HINT: Add or change a related_name argument to the definition for 'base.Notification.action_object_content_type' or 'notifications.Notification.action_object_content_type'.
base.Notification.actor_content_type: (fields.E304) Reverse accessor 'ContentType.notify_actor' for 'base.Notification.actor_content_type' clashes with reverse accessor for 'notifications.Notification.actor_content_type'.
	HINT: Add or change a related_name argument to the definition for 'base.Notification.actor_content_type' or 'notifications.Notification.actor_content_type'.
base.Notification.actor_content_type: (fields.E305) Reverse query name for 'base.Notification.actor_content_type' clashes with reverse query name for 'notifications.Notification.actor_content_type'.
	HINT: Add or change a related_name argument to the definition for 'base.Notification.actor_content_type' or 'notifications.Notification.actor_content_type'.
base.Notification.recipient: (fields.E304) Reverse accessor 'User.notifications' for 'base.Notification.recipient' clashes with reverse accessor for 'notifications.Notification.recipient'.
	HINT: Add or change a related_name argument to the definition for 'base.Notification.recipient' or 'notifications.Notification.recipient'.
base.Notification.recipient: (fields.E305) Reverse query name for 'base.Notification.recipient' clashes with reverse query name for 'notifications.Notification.recipient'.
	HINT: Add or change a related_name argument to the definition for 'base.Notification.recipient' or 'notifications.Notification.recipient'.
base.Notification.target_content_type: (fields.E304) Reverse accessor 'ContentType.notify_target' for 'base.Notification.target_content_type' clashes with reverse accessor for 'notifications.Notification.target_content_type'.
	HINT: Add or change a related_name argument to the definition for 'base.Notification.target_content_type' or 'notifications.Notification.target_content_type'.
base.Notification.target_content_type: (fields.E305) Reverse query name for 'base.Notification.target_content_type' clashes with reverse query name for 'notifications.Notification.target_content_type'.
	HINT: Add or change a related_name argument to the definition for 'base.Notification.target_content_type' or 'notifications.Notification.target_content_type'.
notifications.Notification.action_object_content_type: (fields.E304) Reverse accessor 'ContentType.notify_action_object' for 'notifications.Notification.action_object_content_type' clashes with reverse accessor for 'base.Notification.action_object_content_type'.
	HINT: Add or change a related_name argument to the definition for 'notifications.Notification.action_object_content_type' or 'base.Notification.action_object_content_type'.
notifications.Notification.action_object_content_type: (fields.E305) Reverse query name for 'notifications.Notification.action_object_content_type' clashes with reverse query name for 'base.Notification.action_object_content_type'.
	HINT: Add or change a related_name argument to the definition for 'notifications.Notification.action_object_content_type' or 'base.Notification.action_object_content_type'.
notifications.Notification.actor_content_type: (fields.E304) Reverse accessor 'ContentType.notify_actor' for 'notifications.Notification.actor_content_type' clashes with reverse accessor for 'base.Notification.actor_content_type'.
	HINT: Add or change a related_name argument to the definition for 'notifications.Notification.actor_content_type' or 'base.Notification.actor_content_type'.
notifications.Notification.actor_content_type: (fields.E305) Reverse query name for 'notifications.Notification.actor_content_type' clashes with reverse query name for 'base.Notification.actor_content_type'.
	HINT: Add or change a related_name argument to the definition for 'notifications.Notification.actor_content_type' or 'base.Notification.actor_content_type'.
notifications.Notification.recipient: (fields.E304) Reverse accessor 'User.notifications' for 'notifications.Notification.recipient' clashes with reverse accessor for 'base.Notification.recipient'.
	HINT: Add or change a related_name argument to the definition for 'notifications.Notification.recipient' or 'base.Notification.recipient'.
notifications.Notification.recipient: (fields.E305) Reverse query name for 'notifications.Notification.recipient' clashes with reverse query name for 'base.Notification.recipient'.
	HINT: Add or change a related_name argument to the definition for 'notifications.Notification.recipient' or 'base.Notification.recipient'.
notifications.Notification.target_content_type: (fields.E304) Reverse accessor 'ContentType.notify_target' for 'notifications.Notification.target_content_type' clashes with reverse accessor for 'base.Notification.target_content_type'.
	HINT: Add or change a related_name argument to the definition for 'notifications.Notification.target_content_type' or 'base.Notification.target_content_type'.
notifications.Notification.target_content_type: (fields.E305) Reverse query name for 'notifications.Notification.target_content_type' clashes with reverse query name for 'base.Notification.target_content_type'.
	HINT: Add or change a related_name argument to the definition for 'notifications.Notification.target_content_type' or 'base.Notification.target_content_type'.
pontoon@d0168d836c37:/app$
```

Reply:
Filed https://github.com/mozilla/pontoon/issues/3464.